traptor/traptor.py

```diff
@@ -237,28 +237,56 @@ def _find_rule_matches(self, tweet_dict):
                 new_dict['traptor']['rule_tag'] = rule['tag']
                 new_dict['traptor']['rule_value'] = rule['value']
 
+                for key, value in rule.iteritems():
+                    new_dict['traptor'][key] = value
+
                 return new_dict
 
-        for rule in self.redis_rules:
-            search_str = rule['value']
-            # self.logger.debug("Search string used for the rule match: {}".format(search_str.encode('utf-8')))
-            if re.search(',', search_str):
-                for s in search_str.split(','):
-                    new_dict = self._add_rule_tag_and_value_to_tweet(new_dict, s, rule)
-            else:
-                search_str = rule['value'].split()[0]
-                for i in new_dict.keys():
-                    new_dict = self._add_rule_tag_and_value_to_tweet(new_dict, search_str, rule)
-            # self.logger.debug('Rule matched - tag:{}, value:{}'.format(rule['tag'],
-            #                                                            rule['value'].encode('utf-8')))
-
-            if 'rule_tag' not in new_dict['traptor']:
-                self.logger.warning('Could not find rule_tag: {}, rule_value: {}, in tweet {}'.format(
-                                    rule['tag'], rule['value'].encode('utf-8'), new_dict.get('id_str')))
-                new_dict['traptor']['rule_tag'] = 'Not found'
-                new_dict['traptor']['rule_value'] = 'Not found'
+        if self.traptor_type == 'track':
+
+            for rule in self.redis_rules:
+                # Get the rule to search for and lowercase it
+                search_str = rule['value']
+                search_str = search_str.lower()
+
+                self.logger.debug("Search string used for the rule match: {}".format(search_str.encode('utf-8')))
+                # Lowercase everything in the dict
+                for k, v in new_dict.iteritems():
+                    if v is not None and isinstance(v, str):
```

Comment:
@jasonrhaas @joequery unicode is a a different type than "str" so we need to check for it.